File: HDFC.PDFCoordinateMapper.Api/Models/TemplateMasterModels.cs
namespace HDFC.PDFCoordinateMapper.Api.Models
{
    public sealed class TemplateMasterRequest
    {
        public string Flag { get; set; }
        public string Auto_Id { get; set; }
        public string Mst_Col_Id { get; set; }
        public string Template_Code { get; set; }
        public string Template_Name { get; set; }
        public string Template_Description { get; set; }
        public string Original_File_Name { get; set; }
        public string Stored_File_Name { get; set; }
        public string File_Path { get; set; }
        public string File_Hash { get; set; }
        public string File_Size_Bytes { get; set; }
        public string Mime_Type { get; set; }
        public string Pdf_Page_Count { get; set; }
        public string Mapping_Page_Numbers { get; set; }
        public string Print_Page_Numbers { get; set; }
        public string Repeat_Rows_Per_Page { get; set; }
        public string Is_Digitally_Signed { get; set; }
        public string Digital_Signature_Details { get; set; }
        public string IsActive { get; set; }
        public string CurrentUserId { get; set; }
        public string Remark { get; set; }
    }

    public sealed class TemplateUploadResult
    {
        public string OriginalFileName { get; set; }
        public string StoredFileName { get; set; }
        public string FilePath { get; set; }
        public string FileHash { get; set; }
        public long FileSizeBytes { get; set; }
        public string MimeType { get; set; }
        public int PdfPageCount { get; set; }
    }
}

File: HDFC.PDFCoordinateMapper.WebV1/src/app/features/pdf-coordinate-mapper/template-master/template-master.models.ts
export type TemplateMasterView = 'all' | 'approved';
export type TemplateMasterFormMode = 'create' | 'edit' | 'view';

export interface TemplateMasterRecord {
  raw: Record<string, unknown>;
  autoId: string;
  mstColId: string;
  templateCode: string;
  amcCode: string;
  amcName: string;
  templateName: string;
  templateDescription: string;
  originalFileName: string;
  storedFileName: string;
  filePath: string;
  fileHash: string;
  fileSizeBytes: string;
  mimeType: string;
  pdfPageCount: string;
  mappingPageNumbers: string;
  printPageNumbers: string;
  repeatRowsPerPage: string;
  isDigitallySigned: string;
  digitalSignatureDetails: string;
  active: string;
  status: string;
  action: string;
  actionRemark: string;
  createdBy: string;
  createdDate: string;
  modifiedBy: string;
  modifiedDate: string;
  approvedBy: string;
  approvedDate: string;
}

export interface TemplateUploadResult {
  originalFileName: string;
  storedFileName: string;
  filePath: string;
  fileHash: string;
  fileSizeBytes: number;
  mimeType: string;
  pdfPageCount: number;
}

export interface TemplateMasterFormValue {
  templateCode: string;
  amcCode: string;
  templateName: string;
  templateDescription: string;
  originalFileName: string;
  storedFileName: string;
  filePath: string;
  fileHash: string;
  fileSizeBytes: string;
  mimeType: string;
  pdfPageCount: string;
  mappingPageNumbers: string;
  printPageNumbers: string;
  repeatRowsPerPage: string;
  isDigitallySigned: string;
  digitalSignatureDetails: string;
  active: string;
}

export interface TemplateMasterAmcOption {
  amcCode: string;
  amcName: string;
  raw: Record<string, unknown>;
}

export interface TemplateMasterListSnapshot {
  templates: TemplateMasterRecord[];
  approvedTemplates: TemplateMasterRecord[];
}

export interface TemplateMasterCommandResult {
  success: boolean;
  message: string;
  raw: unknown;
}

export interface TemplateMasterDialogData {
  mode: TemplateMasterFormMode;
  record: TemplateMasterRecord | null;
  submitting: boolean;
}
